packages/runtime-dom/src/apiCustomElement.ts

```diff
@@ -403,9 +403,10 @@ export class VueElement
 
     const asyncDef = (this._def as ComponentOptions).__asyncLoader
     if (asyncDef) {
-      this._pendingResolve = asyncDef().then(def =>
-        resolve((this._def = def), true),
-      )
+      this._pendingResolve = asyncDef().then((def: InnerComponentDef) => {
+        if (this._def.configureApp) def.configureApp = this._def.configureApp
```

Comment:
What do you think about skipping the `if` check?
```suggestion
        def.configureApp = this._def.configureApp
```
It'd save a few bytes and I think it'd be a bit easier to read.